warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ExpandCompositeTerms.java

```diff
@@ -727,18 +800,26 @@ private JexlNode visitLeafNode(JexlNode node, ExpandData eData) {
                 // add any required leaf nodes
                 for (JexlNode node : remainingLeafNodes.get(componentField)) {
                     JexlNode trimmedNode = getLeafNode(node);
-                    if (trimmedNode != null && isNodeValid(trimmedNode, i, componentFields.size(), isFixedLengthField(componentField))) {
-                        nodes.add(trimmedNode);
-                        tempUsedLeafNodes.put(componentField, node);
+                    if (trimmedNode != null) {
+                        if (isNodeValid(trimmedNode, i, componentFields.size(), isFixedLengthField(componentField))) {
+                            nodes.add(trimmedNode);
+                            tempUsedLeafNodes.put(componentField, node);
+                        } else {
+                            log.debug("Node is not valid {} ", JexlStringBuildingVisitor.buildQuery(trimmedNode));
```

Comment:
Same comment/question as above...
```suggestion
                            if (log.isDebugEnabled()) {
                                log.debug("Node is not valid {} ", JexlStringBuildingVisitor.buildQuery(trimmedNode));
                            }
```